baseplate/observers/metrics_tagged.py

```diff
@@ -0,0 +1,187 @@
+import logging
+
+from enum import Enum
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from baseplate import _ExcInfo
+from baseplate import BaseplateObserver
+from baseplate import LocalSpan
+from baseplate import RequestContext
+from baseplate import Span
+from baseplate import SpanObserver
+from baseplate.lib import config
+from baseplate.lib import metrics
+from baseplate.observers.timeout import ServerTimeout
+
+
+logger = logging.getLogger(__name__)
+
+
+class Errors(Enum):
+    TIMED_OUT = 1
+    CONCURRENCY_LIMITED = 2
+    EXCEPTION = 3
+
+
+error_messages = {
+    Errors.TIMED_OUT: "timed_out",
+    Errors.CONCURRENCY_LIMITED: "concurrency_limited",
```

Comment:
ah sorry, i only mentioned this as a heads up this would be coming but it's not necessary to include yet since that patch set isn't fully ready to merge at this time